File: testdata/sample.go
package sample

import "fmt"

/* A block comment that spans
   lines, which the scanner carries. */

const n = 0x1f + 1.5e-3

// Query is a raw string spanning lines, the case a per-line lexer
// colors as code.
const query = `
	SELECT id
	FROM jobs
	WHERE state = 'queued'
`

func main() {
	var s string = "a\"b"
	if c := s[0]; c == '\n' {
		fmt.Println(query, n)
	}
}

File: testdata/sample.js
const s = 1000, m = 60 * s;

/* A block comment that spans
   lines, which the scanner carries. */
function reltime(t) {
	const d = Date.now() - t;
	if (d < 5 * s) return "<5s ago";
	return Math.round(d / s) + "s ago"; // division, not a regex
}

function isField(t) {
	return /^(input|select)$/i.test(t.tagName);
}

const q = `a template literal
that spans lines`;

async function load(el) {
	const r = await fetch(el.dataset.live);
	const open = [...document.querySelectorAll("details")].some((d) => !d.open);
	if (e.key !== "\\") {
		el.innerHTML = await r.text();
	}
	return open;
}
